src/libsystemd/sd-device/sd-device.c

```diff
@@ -1968,20 +1967,17 @@ _public_ int sd_device_set_sysattr_value(sd_device *device, const char *sysattr,
 
         path = prefix_roota(syspath, sysattr);
 
-        len = strlen(_value);
+        value = strdup(_value);
```

Comment:
the old code copied at most 4K, while the new code will copy everything, and then late rcomplain if 4K limit is breached. But I am not sure this matters. this shouldn#t happen IRL, and I like clean easy-to-read code

Reply:
Indeed, the copied string will be cached. I do not want to cache unnecessarily long string. So, the change is dropped.